### PR TITLE
Clean up and improve SPEC file for master branch

### PR DESCRIPTION
**Pull request by anoopcs9**
* Introduce bcond macros for better handling of conditional builds
* Re-arrange and/or split files of internally built libraries(talloc, tevent,tdb and ldb) into _samba-client_, _samba-client-libs_ and _python3-samba_ sub-packages

### CODE REVIEW

File: packaging/samba-master.spec.j2 [hunk 1]
```diff
@@ -598,17 +636,46 @@ Requires: python3-tevent
 Requires: python3-tdb
 Requires: python3-ldb
```
Comment (anoopcs9) [inline packaging/samba-master.spec.j2:637]: May be we can remove above 4 `Requires` as they are provided by the same package.

Comment (gd) [reply packaging/samba-master.spec.j2:638]: Hm, does that take into consideration internal library builds?

Comment (anoopcs9) [reply packaging/samba-master.spec.j2:638]: OK. I think we need those `Requires` anyhow.

**Review by gd — APPROVED** [review]
LGTM, thanks!

**Pull request merged**